lib/tests/streamlit/runtime/caching/cache_data_api_test.py

```diff
@@ -234,6 +234,23 @@ def foo(y):
         foo.clear(1)
         assert foo(1) == 2
 
+    def test_cached_st_method_clear_args(self):
+        self.x = 0
+
+        class ExampleClass:
+            @st.cache_data
+            def foo(_self, y):
+                self.x += y
+                return self.x
+
+        example_instance = ExampleClass()
+
+        assert example_instance.foo(1) == 1
+        example_instance.foo.clear(2)
```

Comment:
nit: Might be worth adding some explanatory comments to describe how this test works just so that it takes a bit less thought for the reader